src/Cli/Microsoft.TemplateEngine.Cli/Commands/create/InstantiateCommand.TabCompletion.cs

```diff
@@ -159,7 +159,7 @@ private static IEnumerable<CliTemplateInfo> GetAllowedTemplates(TemplateConstrai
                 if (constraintEvaluationTask.IsCompletedSuccessfully)
                 {
                     //return only allowed templates
-                    return constraintEvaluationTask.Result;
+                    return Task.Run(() => constraintEvaluationTask).Result;
```

Comment:
Considering three lines up it already checks `containtEvaluationTask.IsCompletedSuccessfully` is true, what this code does now is block the current thread while a new task is scheduled on the thread pool. Once the new task is executed, it is a no-op, and signals back to the original thread that it's ready to continue. Then the operating system has to context switch that thread back into a CPU core.

Basically, this `Task.Run` does nothing more than create memory allocations and add latency, since the task is already complete.